cgen: comparison with == of structs with pointers to interfaces autogenerates invalid eq comparison method

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: linux, Ubuntu 20.04.3 LTS
Processor: 4 cpus, 64bit, little endian, Intel(R) Core(TM) i3-3225 CPU @ 3.30GHz
CC version: cc (Ubuntu 10.3.0-1ubuntu1~20.04) 10.3.0

getwd: /v/vnew
vmodules: /home/delian/.vmodules
vroot: /v/vnew
vexe: /v/vnew/v
vexe mtime: 2022-10-13 16:27:28
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.1 bfbfe78.12b12aa

Git version: git version 2.25.1
Git vroot status: weekly.2022.40-66-gf7f0e0b5 (4 commit(s) behind V master)
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 12f392c3

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg comparing_struct_with_pointers_to_interface_values_should_autogenerate_working_eq_method_test.v`
```v
pub interface Logger {
mut:
	info(s string)
}

struct LogContainer {
	name string
mut:
	logger &Logger = unsafe { nil }
}

fn test_comparing_struct_with_pointers_to_interface_values_autogenerates_working_eq_method() {
	assert LogContainer{
		name: 'abc'
	} == LogContainer{
		name: 'abc'
	}
}

```

**What did you expect to see?**

A compiled and working test.


**What did you see instead?**
```
==================
/tmp/v_1000/comparing_struct_with_pointers_to_interface_values_should_autogenerate_working_eq_method_test.14282662742231591551.tmp.c:3287: error: cannot convert 'struct main__Logger *' to 'struct main__Logger'
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/gen/c/auto_eq_methods.v
+++ b/vlib/v/gen/c/auto_eq_methods.v
@@ -156,8 +156,9 @@ fn (mut g Gen) gen_struct_equality_fn(left_type ast.Type) string {
 			} else if field_type.sym.kind == .function {
 				fn_builder.write_string('*((voidptr*)(a.$field_name)) == *((voidptr*)(b.$field_name))')
 			} else if field_type.sym.kind == .interface_ {
+				ptr := if field.typ.is_ptr() { '*'.repeat(field.typ.nr_muls()) } else { '' }
 				eq_fn := g.gen_interface_equality_fn(field.typ)
-				fn_builder.write_string('${eq_fn}_interface_eq(a.$field_name, b.$field_name)')
+				fn_builder.write_string('${eq_fn}_interface_eq(${ptr}a.$field_name, ${ptr}b.$field_name)')
 			} else {
 				fn_builder.write_string('a.$field_name == b.$field_name')
 			}
@@ -450,7 +451,7 @@ fn (mut g Gen) gen_interface_equality_fn(left_type ast.Type) string {
 					fn_builder.write_string('${eq_fn}_alias_eq(*(a._$eq_fn), *(b._$eq_fn))')
 				}
 				else {
-					fn_builder.write_string('false')
+					fn_builder.write_string('true')
 				}
 			}
 			fn_builder.writeln(';')

--- a/vlib/v/tests/struct_auto_eq_gen_interface_test.v
+++ b/vlib/v/tests/struct_auto_eq_gen_interface_test.v
@@ -21,3 +21,23 @@ fn test_struct_auto_eq_gen_interface_case() {
 	w2 := new()
 	assert w1 == w2
 }
+
+// For https://github.com/vlang/v/issues/16074
+pub interface Logger {
+mut:
+	info(s string)
+}
+
+struct LogContainer {
+	name string
+mut:
+	logger &Logger = unsafe { nil }
+}
+
+fn test_comparing_struct_with_pointers_to_interface_values_autogenerates_working_eq_method() {
+	assert LogContainer{
+		name: 'abc'
+	} == LogContainer{
+		name: 'abc'
+	}
+}

